commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/AbstractFileObject.java

```diff
@@ -1232,6 +1233,7 @@ public FileName getName() {
         return fileName;
     }
 
+    // TODO: remove this method for the next major version as it is unused
```

Comment:
It's a public method, you do not know who uses it, so you can't say it's "unused". I think this comment can be dropped.

Reply:
It's a public method, right. Note, however, that users of VFS always use `FileObject` which is an interface and it doesn't have this method. It's only available if one casts explicitly to `AbstractFileObject` (which no one should do) and to sub-classes (which should not call it at all). The only place this will be called is in `DefaultFileContent` (actually not this but the other overload). That's why I've written that this can be removed - so that in the future a conversation like this is not done. :smiley:

Reply:
Hi @boris-petrov 
It does not matter as `AbstractFileObject` is a public method.

>if one casts explicitly to AbstractFileObject (which no one should do) 

Again, i does not matter as the class is available for 3rd party providers to subclass. Remember that anyone can plugin their own provider(s) in the framework, it's designed to be extensible. There are two kinds of clients of the API: (1) Traditional consumers of the API, and (2) File System providers, like VFS itself.

For 3.0 we can consider breaking the APIs but we are not there yet.

Reply:
You are correct for the two usages. My point is that *neither* customers of the API, *nor* FS providers should be calling these methods. Hence they could be made `final` and package-private. But not before 3.0 as you said. And because we're not there, I've added the TODOs so this is not forgotten.